sql/core/src/test/scala/org/apache/spark/sql/sources/DataSourceAnalysisSuite.scala

```diff
@@ -196,6 +202,7 @@ class DataSourceAnalysisSuite extends SparkFunSuite with BeforeAndAfterAll {
     }
 
     test(s"Static partition and dynamic partition (caseSensitive: $caseSensitive)") {
+      SQLConf.get.setConf(SQLConf.CASE_SENSITIVE, caseSensitive)
```

Comment:
I have a better idea. We can create a `test` method in this suite:
```
def test(caseSensitive: Boolean, testName: String)(func: => Unit) = {
  test(s"$testName (caseSensitive: $caseSensitive)") {
    withSQLConf(case sen....) {
      func
    }
  }
}
```

Reply:
Then we don't need to change `beforeAll/afterAll`, and it's one line diff for each test.

Reply:
done